packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -103,7 +103,20 @@ contract SignalService is ISignalService, EssentialContract {
     function getSignalSlot(
         address app,
         bytes32 signal
-    ) public pure returns (bytes32) {
-        return keccak256(abi.encodePacked(app, signal));
+    ) public pure returns (bytes32 signalSlot) {
+        assembly {
```

Comment:
@cyberhorsey  could you apply the same optimization to the AddressManager's key calculation?

Reply:
I meant "AddressResolver"'s key

Reply:
Will do